src/conditions/requiredAuthorRole.ts

```diff
@@ -0,0 +1,20 @@
+import { ConditionConfig } from './../config'
+import { PullRequestInfo } from '../models'
+import { ConditionResult } from '../condition'
+import { getAssociationPriority } from '../association'
+
+export default function hasRequiredAuthorRole (
+  config: ConditionConfig,
+  pullRequestInfo: PullRequestInfo
+): ConditionResult {
+  const associationString = ['NONE', 'FIRST_TIME', 'FIRST_TIME_CONTRIBUTOR', 'CONTRIBUTOR', 'COLLABORATOR', 'MEMBER', 'OWNER']
+
+  return associationString.indexOf(config.requiredAuthorRole) > getAssociationPriority(pullRequestInfo.authorAssociation)
```

Comment:
Not being able to use `getAssociationPriority` is because the config is not typed correctly. See below in `config.ts`.

Reply:
Added a commit for this now that the type of `requiredAuthorRole` is `CommentAuthorAssociation`.